src/olympia/abuse/management/commands/auto_resolve_reports.py

```diff
@@ -0,0 +1,60 @@
+from collections import defaultdict
+
+from django.core.management.base import BaseCommand
+
+import olympia.core.logger
+from olympia.abuse.models import AbuseReport, CinderJob
+from olympia.abuse.tasks import handle_already_moderated
+from olympia.versions.models import Version
+
+
+class Command(BaseCommand):
+    log = olympia.core.logger.getLogger('z.abuse')
+
+    def handle(self, *args, **options):
+        jobs = (
+            CinderJob.objects.unresolved()
+            .resolvable_in_reviewer_tools()
+            .select_related('target_addon___current_version')
+            .prefetch_related('appealed_decisions')
+        )
+        abuse_report_dict = defaultdict(list)
+        abuse_reports = list(
+            AbuseReport.objects.filter(cinder_job__in=jobs).values_list(
+                'reason', 'addon_version', 'cinder_job_id', named=True
+            )
+        )
+        for report in abuse_reports:
+            abuse_report_dict[report.cinder_job_id].append(report)
+
+        addon_versions = list(
+            # Note this query can over-match Version objects if the version string
+            # exists in multiple add-ons (e.g. version "0.1" exists in addon-A and
+            # addon-B, but is not reported for addon-A, only for addon-B)
+            # - this is somewhat inefficient, but harmless - it just won't be looked
+            # up in CinderJob.should_auto_resolve.
+            Version.objects.filter(
+                addon__in=[job.target_addon for job in jobs],
+                version__in=[
+                    report.addon_version
+                    for report in abuse_reports
+                    if report.addon_version
+                ],
```

Comment:
At a minimum this should be a set, not a list of versions.

I tried running the `addon_versions` queryset in prod. The `version__in` part causes the query to take forever to execute, I had to abort it after 5 minutes. There are a lot of reports... I wasn't running 100% the exact same thing, I had to use subqueries for `addon__in` and `version__in`, but still, I'm concerned it would not execute properly.

It might be better to grab affected versions individually for each job in the loop later rather than trying to prefetch them all in one go...